lms/djangoapps/courseware/courses.py

```diff
@@ -137,26 +147,69 @@ def get_course_overview_with_access(user, action, course_key, check_if_enrolled=
         course_overview = CourseOverview.get_from_id(course_key)
     except CourseOverview.DoesNotExist:
         raise Http404("Course not found.")
-    check_course_access(course_overview, user, action, check_if_enrolled)
+    check_course_access_with_redirect(course_overview, user, action, check_if_enrolled)
```

Comment:
All existing calls to check_course_access need to be sent to check_course_access_with_redirect instead to preserve functionality.

Reply:
I agree on refactoring to separate the access check from the redirect. And I agree that the names are rational with this respect. But it does break backwards compatibility for the things that might call this method from outside edx-platform (for instance enterprise, or plugin apps). So I'd ask that either:

1. We keep the default behavior of `check_course_access` and add a new method that does not have the redirect.
2. We document that this behavior changed in the [Juniper release notes](https://openedx.atlassian.net/wiki/spaces/COMM/pages/940048716/Juniper) page. (my preference)

Reply:
(FWIW, I did check edx-enterprise and they don't use it. But there's a bunch of stuff out there that we won't see, and that we encourage to get installed in-proc with Django app plugins.)